backend/api/badge_scanner/schema.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import annotations
+
+import pretix
+import re
+from typing import Any
+import strawberry
+from strawberry.types.info import Info
+from typing import Optional, Union
+
+from api.permissions import IsAuthenticated
+from badge_scanner import models
+from conferences.models import Conference
+from users.client import get_user_by_email
+
+
+@strawberry.type
+class ScanError:
+    message: str
+
+
+@strawberry.input
+class ScanBadgeInput:
+    url: str
+    conference_code: str
+
+    url_regex = re.compile(r"^https://pycon\.it/b/([\w\d-]+)$")
+
+    def validate(self) -> ScanError | None:
+        if not self.url:
+            return ScanError(message="URL is required")
+
+        match = self.url_regex.match(self.url)
+
+        return None if match else ScanError(message="URL is not valid")
+
+    @property
+    def order_position_id(self) -> str:
+        match = self.url_regex.match(self.url)
+
+        assert match
+
+        return match.group(1)
+
+
+@strawberry.type
+class Attendee:
+    full_name: str
+    email: str
+
+
+@strawberry.type
+class BadgeScan:
+    attendee: Attendee
+    notes: Optional[str]
+
+
+@strawberry.type
+class BadgeScannerMutation:
+    @strawberry.mutation(
+        permission_classes=[IsAuthenticated],
+    )
+    def scan_badge(
+        self, info: Info[Any, None], input: ScanBadgeInput
+    ) -> Union[BadgeScan, ScanError]:
```

Comment:
since you added the future import and are using it above, you should use the pipe operator here as well to be consistent

```suggestion
    ) -> BadgeScan | ScanError:
```